Cython/Debugger/Cygdb.py

```diff
@@ -104,42 +104,36 @@ def make_command_file(path_to_debug_info, prefix_code='',
 
     return tempfilename
 
-usage = "Usage: cygdb [options] [PATH [-- GDB_ARGUMENTS]]"
 
-def main(path_to_debug_info=None, gdb_argv=None, no_import=False):
+def main():
     """
     Start the Cython debugger. This tells gdb to import the Cython and Python
     extensions (libcython.py and libpython.py) and it enables gdb's pending
     breakpoints.
-
-    path_to_debug_info is the path to the Cython build directory
-    gdb_argv is the list of options to gdb
-    no_import tells cygdb whether it should import debug information
     """
-    parser = optparse.OptionParser(usage=usage)
-    parser.add_option("--gdb-executable",
+    parser = argparse.ArgumentParser(
+        prog="cygdb",
+        description="Cython debugger",
+    )
+    parser.add_argument("gdb_argv", nargs="*",
+                        help="Arguments to forward to gdb; specified after --")
+    parser.add_argument("--build-dir", dest="build_dir", default=None,
+                        help="Directory containing cython_build/ files")
```

Comment:
Does this argument give the directory where the `cython_build/*` *files* live (i.e. `some/path/cython_build/`) or the directory where the `cython_build/` *directory* lives? What is a `cython_build` directory anyway? Was the `cython_debug` directory meant, maybe? I'm really confused by the name of the argument and its description.

Reply:
It essentially replaces the old argument of `path_to_debug_info` - would you prefer to revert back to that name?

Reply:
Since it's a generated directory, which should usually go into the `build/` directory, `--build-dir` seems appropriate. The question is then whether we really write the `debug_info` directory into the `build/` directory, and it seems that we don't but rather write it into the current directory instead. If that's the build directory, that's fine, otherwise, it'd probably be the project root.

Should we move the `debug_info` directory into `build/`, unless explicitly set otherwise? There's probably a way to extract the build directory from something like `setuptools` if that's used, but if we're called via command line, that won't be available/interesting/correct.

Reply:
If anyone knows how to query for that information it is probably @rgommers 

Ralf - do you know if there's a way to query the build (frontend? backend?) to know what the "builddir" setting is?

Reply:
(disclaimer: I'm answering with probably a bit of context lacking here)

> Ralf - do you know if there's a way to query the build (frontend? backend?) to know what the "builddir" setting is?

My understanding is that `cygdb` is invoked as a standalone tool, sometime well after a build was completed. At that point there is no build frontend or build backend to query. Also, `cygdb` can be executed anywhere, rather than it having to be run from a project root, so there's no good way to even guess.

The current invocation (from [the cygdb docs](https://cython.readthedocs.io/en/latest/src/userguide/debugging.html#running-the-debugger)) will be something like:
```
cygdb path/to/builddir -- --args python-dbg mainscript.py
```
and this PR replaces the absolute argument so `path/to/builddir` would become `--build-dir path/to/builddir`? Either way I don't quite see a way around the user having to provide the location of the build dir.



> Should we move the `debug_info` directory into `build/`, unless explicitly set otherwise?

I don't think so, since the `cython` executable is typically invoked from within the build directory already - at least with Meson/CMake/ninja - so if you'd make this change then it would land in `build/build/`.